### PR TITLE
Redact authTokens when logging an SQLite error

### PR DESCRIPTION
**Pull request by joelbettner**
### Details
This PR make sure we do not log an auth token when logging an SQLite error.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/266804

### Tests
1. Set up the `Notifications::add` method in Auth to create a JSON that was malformed.
2. Called `CreateNotification` with netcat that had a JSON with an authToken in it (don't worry, this authToken is from a fake account run locally):
```
vagrant@expensidev2004:/vagrant/Auth$ nc localhost 4444
CreateNotification
authToken: 5A59AC4730E31662263FEC022A3097076E2D0EAE9494BF56410B4EFEA9E1109FB8B2E79CBE198E08282F01290CEB0C1157251771A6B34AB4027312C6E30ED2429BDE0A2E2F8D13788C3D3A86576E9D77CE83959AA6197BF3C18A926CFD33188553530C55DA51ECD770DA835C3AA6E3CE32047082EAFE5CB0F7A83CAD98916D2E1215FDDFE8257DDD36BD1C328B8A9C96ACA0C08503CC11697FAD81CDB40B793EC167EEA27C8AE35C3C29337FB9F416F98A2DB75DEB3B569BB668B364F0BEEA94AB1266A6DFB9289CAC123B9CA65E4599DAF248C6A77BF31F213450F5B70311098E243FB97E21DE8A066AB45B9E5A85432A9DF969DD7EEA313B87E5F1462628C02E919146A46EF16D0E317615A907930EA91800A382264FB5EF143A0D1AF7587FD2C53D16F0C41185FFB33D289E01B297C76450746D246CB09B1A85A13E5A58206360B1BFFAA0314C6FDD89FF9D06F3EF823153B15AE842AB9D349E69E0A6B13E
From: Expensify Concierge <concierge@expensify.com>
reply-to: concierge@expensify.com
to: jbexpensifytest+2@gmail.com
title: Expensify Integrations to QuickBooks Online Warning (Report ID: 80805898)
message: 2023-03-07 22:53:46::752652: Request ID jnLS67#0122023-03-07 22:53:46::752705: User chelsea@cinespia.org attempted to export report [80805898] - 2021-09-23 - 2021-09-23#0122023-03-07 22:53:46::752722: Export is to QuickBooks Online#0122023-03-07 22:53:46::752738: <span style="color:red;">Oops! Expensify encountered an error when auto-exporting to QuickBooks Online, and this requires your attention.</span>#0122023-03-07 22:53:46::752753: <span style="color:red;">Account Event Add-Ons:Dance Floor not found on policy E5ABEAC9AD1F1762, make sure the QBO connection is synced.</span>#012
integrationName: QuickBooks Online
template: Integrations_Warning

502 Query failed
commitCount: 68016
nodeName: auth
peekTime: 1552
processTime: 10881
requestID: 6MUVdo
totalTime: 13036
unaccountedTime: 603
Content-Length: 0
```
3. Confirmed that the auth command was redacted from the SQLite error:
image
```
2023-03-09T01:32:54.410940+00:00 expensidev2004 bedrock: 4SyFz6 (SQLite.cpp:262) _sqliteLogCallback [socket0] [info] {SQLITE} Code: 1, Message: statement aborts at 5: [INSERT INTO notifications (json, priority, created) VALUES ( JSON('{"accountID":15,"authToken":"<REDACTED>"
2023-03-09T01:32:54.410988+00:00 expensidev2004 bedrock: 4SyFz6 (libstuff.cpp:2664) SQuery [socket0] [warn] 'read/write transaction', query failed with error #1 (malformed JSON): INSERT INTO notifications (json, priority, created) VALUES ( JSON('{"accountID":15,"authToken":<REDACTED>,"From":"Expensify Concierge <concierge@expensify.com>","integrationName":"QuickBooks Online","message":"2023-03-07 22:53:46::752652: Request ID jnLS67#0122023-03-07 22:53:46::752705: User chelsea@cinespia.org attempted to export report [80805898] - 2021-09-23 - 2021-09-23#0122023-03-07 22:53:46::752722: Export is to QuickBooks Online#0122023-03-07 22:53:46::752738: <span style=\"color:red;\">Oops! Expensify encountered an error when auto-exporting to QuickBooks Online, and this requires your attention.<\/span>#0122023-03-07 22:53:46::752753: <span style=\"color:red;\">Account Event Add-Ons:Dance Floor not found on policy E5ABEAC9AD1F1762, make sure the QBO connection is synced.<\/span>#012","reply-to":"concierge@expensify.com","template":"Integrations_Warning","title":"Expensify Integrations to QuickBooks Online Warning (Report ID: 80805898)","to":"jbexpensifytest+2@gmail.com"}}'), 100, '2023-03-09 01:32:54');
```

### CODE REVIEW

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]
I think it makes more sense to do this once here:
https://github.com/Expensify/Bedrock/blob/main/libstuff/libstuff.h#L236

Rather than adding the same redaction logic in multiple functions and files.

**Comment by joelbettner** [comment]
It might be a bit much to regex search every log line for an auth token...no?

**Comment by tylerkaraszewski** [comment]
What if we extract the redaction code from here: https://github.com/Expensify/Bedrock/blob/main/libstuff/libstuff.cpp#L2622-L2628

And move it into it's own function, and call that from the several places we need to redact this stuff?

File: libstuff/libstuff.cpp [hunk 1]
```diff
@@ -2760,6 +2753,18 @@ bool SREMatch(const string& regExp, const string& s, string& match) {
     return pcrecpp::RE(regExp).FullMatch(s, &match);
 }
 
+void redactredactSensitiveValues(string& s) {
```
Comment (tylerkaraszewski) [inline libstuff/libstuff.cpp:2756]: This should probably be called `SRedactSensitiveValues`. It should definitely not be called `redactredactSensitiveValues` 😆

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]

File: libstuff/libstuff.cpp (hunk 1 above)
Comment (joelbettner) [reply libstuff/libstuff.cpp]: I don't know whatwhat you mean. 😆

**Comment by joelbettner** [comment]
@tylerkaraszewski Updated. Let me know what you think.

File: sqlitecluster/SQLite.cpp [hunk 2]
```diff
@@ -256,7 +256,8 @@ int SQLite::_walHookCallback(void* sqliteObject, sqlite3* db, const char* name,
 
 void SQLite::_sqliteLogCallback(void* pArg, int iErrCode, const char* zMsg) {
     _mostRecentSQLiteErrorLog = "{SQLITE} Code: "s + to_string(iErrCode) + ", Message: "s + zMsg;
-    SSYSLOG(LOG_INFO, "[info] " << _mostRecentSQLiteErrorLog);
+    SRedactSensitiveValues(_mostRecentSQLiteErrorLog);
```
Comment (tylerkaraszewski) [inline sqlitecluster/SQLite.cpp:259]: I think that this should just be part of the previous line: `SRedactSensitiveValues(zMsg)`

Edit: OK, I see why it isn't, the argument is modified in place. We can eave it as-is.

File: sqlitecluster/SQLite.cpp [hunk 3]
```diff
@@ -256,7 +256,8 @@ int SQLite::_walHookCallback(void* sqliteObject, sqlite3* db, const char* name,
...
+    SINFO(_mostRecentSQLiteErrorLog);
```
Comment (tylerkaraszewski) [inline sqlitecluster/SQLite.cpp:260]: This seems fine, I'm trying to think of any reason it would have been the way it was, except for an oversight.

**Review by tylerkaraszewski — APPROVED** [review]

**Pull request merged**